File: backend/src/Features/Contacts/Dto/ContactDetailsDto.cs
namespace RecruitmentApp.Features.Contacts.Dto;

public class ContactDetailsDto
{
    public int Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Email { get; set; }
    public required string PhoneNumber { get; set; }
    public required string Password { get; set; }
    public DateOnly DateOfBirth { get; set; }
    public required CategoryDto CategoryDto { get; set; }
    public SubcategoryDto? SubcategoryDto { get; set; }
}

File: frontend/src/app/modules/contacts/models/contact-details-dto.ts
import { CategoryDto } from './category-dto';
import { SubcategoryDto } from './subcategory-dto';

export interface ContactDetailsDto {
  readonly id: string;
  readonly firstName: string;
  readonly lastName: string;
  readonly email: string;
  readonly phoneNumber: string;
  readonly password: string;
  readonly dateOfBirth: string;
  readonly category: CategoryDto;
  readonly subcategory: SubcategoryDto;
}
